cloud/scope/powervs_cluster.go

```diff
@@ -1744,24 +1753,29 @@ func (s *PowerVSClusterScope) fetchResourceGroupID() (string, error) {
 	return "", err
 }
 
-// getVPCRegion returns region associated with VPC zone.
-func (s *PowerVSClusterScope) getVPCRegion() *string {
-	if s.IBMPowerVSCluster.Spec.VPC != nil {
-		return s.IBMPowerVSCluster.Spec.VPC.Region
-	}
-	// if vpc region is not set try to fetch corresponding region from power vs zone
+// getTransitGatewayLocationAndRouting returns appropriate location and routing suitable for transit gateway.
+// routing indicates whether to enable global routing on transit gateway or not.
+// returns true when powervs and vpc region are not common otherwise false.
+func (s *PowerVSClusterScope) getTransitGatewayLocationAndRouting() (*string, *bool) {
```

Comment:
I dont see any advantage of using pointer to bool , I think we can simply use bool instead.

Reply:
Anyway I need to make it a pointer in called func, so just made it here and used as it is.

Reply:
that makes sense.